src/components/_internal/BaseSingleDatePicker/CustomHeader.tsx

```diff
@@ -0,0 +1,86 @@
+import styled from 'styled-components';
+import React from 'react';
+
+import { PaddingTypes } from '../../layout/Padbox/Padbox.enums';
+import { SpaceSizes } from '../../..';
+import { Icon, Inline, Padbox } from '../..';
+import { SSCIconNames } from '../../../theme/icons/icons.enums';
+import { getColor, getRadii } from '../../../utils';
+import { ColorTypes } from '../../../theme/colors.enums';
+
+const months = [
+  'January',
+  'February',
+  'March',
+  'April',
+  'May',
+  'June',
+  'July',
+  'August',
+  'September',
+  'October',
+  'November',
+  'December',
+];
+
+const getYear = (date: Date) => date.getFullYear();
+const getMonth = (date: Date) => months[date.getMonth()];
+
+const ChangeMonthIconContainer = styled.div<{ direction: string }>`
+  transform: ${(props) =>
+      props.direction === 'right' ? 'rotate(270deg)' : 'rotate(90deg)'}
+    translate(0, -4px);
+  font-size: 16px;
+  width: 32px;
+  height: 32px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+`;
+
+const MonthSelector = styled.button`
+  border-radius: ${getRadii('default')};
+  &:hover {
+    background-color: ${getColor(ColorTypes.primary50)};
+  }
+  &:hover svg {
+    color: ${getColor(ColorTypes.primary600)};
+  }
+`;
+
+const ChangeMonthIcon = (props) => (
+  <ChangeMonthIconContainer {...props}>
+    <Icon color={ColorTypes.neutral600} name={SSCIconNames.sortDown} />
+  </ChangeMonthIconContainer>
+);
+
+const MonthIndicator = styled.button`
+  border-radius: ${getRadii('default')};
+  &:hover {
+    color: ${getColor(ColorTypes.primary600)};
+    background-color: ${getColor(ColorTypes.primary50)};
+  }
+`;
+
+export const DatePickerCustomHeader: React.FC<{
+  decreaseMonth: () => void;
+  increaseMonth: () => void;
+  toggleYearPicker: () => void;
+  date: Date;
+}> = ({ decreaseMonth, toggleYearPicker, date, increaseMonth }) => {
+  return (
+    <Inline justify="space-between">
+      <MonthSelector type="button" onClick={decreaseMonth}>
```

Comment:
It would be great if we can list through years in the year picker mode. There are props `increaseYear` and `decreaseYear` provided that we can use for this.
```suggestion
      <MonthSelector type="button" onClick={isYearPickerOpen ? decreaseYear : decreaseMonth}>
```